plugins/remember.py

```diff
@@ -104,14 +105,62 @@ def forget(inp, chan='', db=None):
         return "I don't know about that."
 
 
-@hook.regex(r'^\? ?(.+)')
+def get_page(data, start_idx, min_page_len, max_page_len):
+    page_data = data[start_idx:start_idx + max_page_len]
+    last_comma_idx = page_data.rfind(',')
+
+    if last_comma_idx == -1 or last_comma_idx < min_page_len:
+        ret = data[start_idx:max_page_len + start_idx]
+        return ret, len(ret) + start_idx
+
+    end_idx = last_comma_idx + start_idx
+    return data[start_idx:end_idx], end_idx
+
+
+def get_pages(data, min_page_len, max_page_len):
+    result = []
+
+    page_data, last_idx = get_page(data, 0, min_page_len, max_page_len)
+    while page_data != '':
+        result.append(page_data)
+        page_data, last_idx = get_page(data, last_idx, min_page_len, max_page_len)
+
+    return result
+
+
+@hook.regex(r'^\? ?(\S+) ?(\d+)?')
 def question(inp, chan='', say=None, db=None):
-    "?<word> -- shows what data is associated with word"
+    "?<word> <page?>-- shows what data is associated with word"
     db_init(db)
 
-    data = get_memory(db, chan, inp.group(1).strip())
+    more_message = ' (%s page(s) left)'
+    message_len_limit = 512 - len(more_message) - 75 # fudge factor for protocol overhead
+    min_page_len = 100
+
+    word = inp.group(1)
+    page = inp.group(2)
+
+    if page is None:
+        page = 1
+
+    page = int(page)
+
+    data = get_memory(db, chan, word.strip())
     if data:
-        say(data)
+        pages = get_pages(data, min_page_len, message_len_limit)
+        page_idx = page - 1 # 1-indexed
+        try:
+            page_data = pages[page_idx]
+        except IndexError:
+            return
```

Comment:
I figured the precedent was not to, for example trying to retrieve a missing remember has no error response -- but we can do like "<word> only has 3 pages"?

@rmmh